Missing `build-report` makefile target on Windows

[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

The make target `build-report` is missing on Windows and needs to be backported
from the *nix makefile target.

## Steps to Reproduce

Run this on Windows:
```
PS C:\relax\couchdb> make -f Makefile.win build-report
```
Output:
```
make: *** No rule to make target 'build-report'.  Stop.
```

## Expected Behaviour

Should run the test like the makefile target on *nix:
```
build-aux/show-test-results.py --suites=10 --tests=10 > test-results.log
```

## Your Environment

* CouchDB version used: a14922fad42575574bb26a24a123e0a1ef4e1621
* Operating system and version: Windows Server 2022

--- a/Makefile.win
+++ b/Makefile.win
@@ -250,6 +250,11 @@ elixir-check-formatted: elixir-init
 elixir-credo: elixir-init
 	@mix credo
 
+.PHONY: build-report
+# target: build-report - Generate a build report
+build-report:
+	@$(PYTHON) build-aux/show-test-results.py --suites=10 --tests=10 > test-results.log
+
 .PHONY: check-qs
 # target: check-qs - Run query server tests (ruby and rspec required!)
 check-qs:

